### PR TITLE
Code cleanup & optim

### PR DESCRIPTION
**Pull request by thomas-kl1**
The intent of this PR is to fix many issues:

- if the Amasty_Base is not installed, an error 500 is thrown on the backend because the menu removed is not declared previously
- code quality by reduci'g LLOC and complexity
- remove the advertising code which add extra loaded code and files and is not relevant regarding the module achievement goal (also seen in #19 
- support for Magento 2.4
- remove code depreciation

### CODE REVIEW

**Comment by rav-redchamps** [comment]
Hi @thomas-kl1 

Thanks for taking out time to prepare this pull request & sorry for the delay in response.

I will be checking & merging it by this week if all found to be okay.

File: view/adminhtml/web/js/menu.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
-require([
```
Comment (rav-redchamps) [inline view/adminhtml/web/js/menu.js]: @thomas-kl1 Not sure if this file is knowingly removed. This code was introduced in recent release to have a small non interrupting "powered by" link.

There is already a feedback/issue for this https://github.com/redchamps/clean-admin-menu/issues/19 but we are still evaluating if we should keep it or revert the change.

Comment (thomas-kl1) [reply view/adminhtml/web/js/menu.js]: Yes I removed the file on purpose. I don't make sense ta have such link regarding the purpose of the module.

Comment (rav-redchamps) [reply view/adminhtml/web/js/menu.js]: Make sense. But we have added it for some purpose & made it as non-interrupting as possible :-) So that it doesn't create any usability problem to merchants.

However, we are still evaluating if this change can be reverted.

Comment (thomas-kl1) [reply view/adminhtml/web/js/menu.js]: That's vszdnbecausebat the end the Magento looks like a Christmas tree. Adding new menu entries does not affect usability big merchants then.
I think adding code that does not have any value for merchant is a loss of quality, performance and is not eco-friendly

Comment (rav-redchamps) [reply view/adminhtml/web/js/menu.js]: Absolutely, it has no benefit for the merchant & it should be removed from the merchant perspective. But from our perspective, it serves a value to us & doesn't impact merchant much logically.

But please allows us some time to figure this out, probably we will go ahead to remove it or look for some alternative approach.

**Comment by rav-redchamps** [comment]
Hi @thomas-kl1 

Very sorry for the delay.

I am testing your changes and noticed that the pull requests breaks compatibility with 2.3.x versions. I tried installing it on 2.3.5-p1. 

 Problem 1
    - magento/framework 103.0.0 requires laminas/laminas-code ~3.4.1 -> no matching package found.
    - magento/framework 103.0.0-p1 requires laminas/laminas-code ~3.4.1 -> no matching package found.
    - magento/framework 103.0.1 requires laminas/laminas-code ~3.4.1 -> no matching package found.
    - redchamps/module-clean-admin-menu dev-master requires magento/framework ^103.0 -> satisfiable by magento/framework[103.0.1, 103.0.0-p1, 103.0.0].
    - Installation request for redchamps/module-clean-admin-menu dev-master -> satisfiable by redchamps/module-clean-admin-menu[dev-master].

The error is due to constraints mentioned in composer.json

File: etc/adminhtml/menu.xml [hunk 2]
```diff
@@ -9,7 +9,5 @@
     <menu>
         <add id="RedChamps_CleanMenu::extensions" title="Extensions" translate="title" module="RedChamps_CleanMenu" sortOrder="60" dependsOnModule="RedChamps_CleanMenu" resource="RedChamps_CleanMenu::extensions"/>
         <update id="Magento_Marketplace::partners" dependsOnConfig="clean_admin_menu/marketplace/enabled" dependsOnModule="Magento_Marketplace" title="Find Partners &amp; Extensions" translate="title" resource="Magento_Marketplace::partners"/>
-        <!--This is to fix conflict with Amasty_Base Extension-->
-        <remove id="Amasty_Base::marketplace" />
```
Comment (rav-redchamps) [inline etc/adminhtml/menu.xml]: I have tried extension installation on clean Magento 2.4.1 instance and it doesn't trigger any error. Also, we haven't received any complaint about this problem.

**Comment by thomas-kl1** [comment]
Hi, yes it breaks with M2.3, I get rid of deprecated usage such as the escape functions in the blocks

**Comment by rav-redchamps** [comment]
> Hi, yes it breaks with M2.3, I get rid of deprecated usage such as the escape functions in the blocks

Thanks for the response. But in ideal case. We should keep it compatible with at least three recent release lines 2.2.x, 2.3.x & 2.4.x 

People are still using 2.2.x & 2.3.x. There are very less stores on 2.4.x (due to more server requirements like compulsion of Elasticsearch)

File: etc/adminhtml/menu.xml (hunk 2 above)
Comment (thomas-kl1) [reply etc/adminhtml/menu.xml]: this breaks on M2.2

**Comment by thomas-kl1** [comment]
I've fixed the breaking changes and fixed merge conflict. I've also reverted the changes to the powered link regarding #19 and 219a001a5156dc39e2663d3ca0a2e538dbeec285

File: etc/adminhtml/menu.xml (hunk 2 above)
Comment (thomas-kl1) [reply etc/adminhtml/menu.xml]: M2.2 is eol so I agree let's ignore this, I've reverted this change.

File: Plugin/Model/MenuBuilder.php [hunk 3]
```diff
@@ -15,21 +15,12 @@ final class MenuBuilder
 {
     public function afterGetResult(Builder $subject, Menu $result): Menu
     {
-        $shouldRemoveMenu = false;
```
Comment (rav-redchamps) [inline Plugin/Model/MenuBuilder.php:18]: This code/function was in place to remove the "Extensions" menu item if there is not child items under it. 

After the changes in pull request to this code, broke this behaviour. May you please give it a check.

Comment (thomas-kl1) [reply Plugin/Model/MenuBuilder.php:18]: I can't reproduce the issue, please could you give some inputs?
If you look the code, the menu is removed if there is no items/no allowed items.

Comment (thomas-kl1) [reply Plugin/Model/MenuBuilder.php:18]: @rav-redchamps were you able to reproduce the error?

**Comment by onlinebizsoft** [comment]
@rav-redchamps can you check asap and accept this pull request? The latest release doesnt work with 2.4 which is popular now.

**Comment by rav-redchamps** [comment]
@onlinebizsoft sure, will do. May you please share the issue you are facing with 2.4. This will allow us to make sure pull request solves the specific issue.

File: Plugin/Model/MenuBuilder.php (hunk 3 above)
Comment (rav-redchamps) [reply Plugin/Model/MenuBuilder.php:18]: Hey @thomas-kl1 

Very sorry for long delay on this. I will be checking this today/tomorrow & get back to you.
